### PR TITLE
DATACOUCH-604 - Fix NPE when looking for Id field in interface.

### PR DESCRIPTION
**Pull request by mikereiche**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATACOUCH).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 
 import com.couchbase.client.core.deps.com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.security.acl.LastOwnerException;
```
Comment (daschl) [inline src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java:31]: unneeded import?

**Review by daschl — CHANGES_REQUESTED** [review]
Can we add  a test case as a regression test for this npe?

File: src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java [hunk 2]
```diff
@@ -99,7 +100,13 @@ public String getFieldName() {
 	// DATACOUCH-145: allows SDK's @Id annotation to be used
 	@Override
 	public boolean isIdProperty() {
-		return isAnnotationPresent(Id.class) || super.isIdProperty()
-				|| this.getFieldName().toLowerCase(Locale.ROOT).equals("id");
+		if (isAnnotationPresent(Id.class) || super.isIdProperty()){
```
Comment (mp911de) [inline src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java:103]: This check can be simplified to:

```java
		if (super.isIdProperty()) {
			return true;
		}
```

File: src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java [hunk 3]
```diff
@@ -99,7 +100,13 @@ public String getFieldName() {
...
+			return true;
+		}
+		// is field named "id"
+		if(getField() != null && this.getFieldName().toLowerCase(Locale.ROOT).equals("id")){
```
Comment (mp911de) [inline src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java:107]: In MongoDB, we simplify that check to:

```
SUPPORTED_ID_PROPERTY_NAMES.contains(getName()) && !hasExplicitFieldName()
```

Here, you could check for e.g. `"id".equals(getName())`.

**Review by mp911de — COMMENTED** [review]
I left two comments how the check could be simplified. Especially `isAnnotationPresent(…)` shouldn't be necessary.

File: src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java (hunk 3 above)
Comment (mikereiche) [reply src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java:107]: Done. Thanks!

File: src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java (hunk 2 above)
Comment (mikereiche) [reply src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java:103]: Done. Thanks!

File: src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java (hunk 1 above)
Comment (mikereiche) [reply src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java:31]: Removed.  I have no idea.

**Review by daschl — APPROVED** [review]

**Pull request closed without merging**